app/main/authorize/access_token_sign_in_required.py

```diff
@@ -40,10 +41,25 @@ def decorated_view(*args, **kwargs):
         g.access_token_sign_in_required = True  # Set attribute on g
 
         try:
-            if not _validate_access_token():
+            access_token = session["access_token"]
+            refresh_token = session["refresh_token"]
+            try:
+                (
+                    access_token,
+                    refresh_token,
+                    tokens_are_refreshed,
+                ) = _validate_or_refresh_tokens(access_token, refresh_token)
+            except InvalidAccessToken:
                 session.clear()
                 return redirect(url_for("main.sign_in"))
 
+            if tokens_are_refreshed:
```

Comment:
no point to update the session if they werent refreshed, as theyd be the same values as before

Reply:
![image](https://github.com/nationalarchives/da-ayr-beta-webapp/assets/107114665/5794972c-9831-4fab-8b73-229f41ca4987)